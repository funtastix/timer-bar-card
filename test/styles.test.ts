import { multiply, HomeAssistant, PlaywrightBrowser, PlaywrightElement } from "hass-taste-test";
import { toMatchDualSnapshot, waitForTimerTime } from "./util";

expect.extend({ toMatchDualSnapshot });

const CONFIGURATION_YAML = `
timer:
  ${multiply(10, (i) => `
  test${i}:
    duration: "00:${i}:00"
`)}
`;

let hass: HomeAssistant<PlaywrightElement>;

beforeAll(async () => {
  hass = await HomeAssistant.create(CONFIGURATION_YAML, {
    browser: new PlaywrightBrowser(process.env.BROWSER || "firefox"),
  });
  await hass.addResource(__dirname + "/../dist/timer-bar-card.js", "module");
}, 30000);
afterAll(async () => await hass.close());

it("Widths", async () => {
  const dashboard = await hass.Dashboard([
    {
      type: "custom:timer-bar-card",
      name: "Progress bar should be munched",
      entities: ["timer.test1"],
      bar_width: "50px",
      text_width: "10px",
    },
  ]);
  const card = dashboard.cards[0];
  await expect(card).toMatchDualSnapshot("idle");

  await hass.callService("timer", "start", { duration: "10:00:00" }, { entity_id: "timer.test1" });
  await expect(card).toMatchDualSnapshot("running");
});

it("Colors and icons", async () => {
  const dashboard = await hass.Dashboard([
    {
      type: "custom:timer-bar-card",
      bar_height: "20px",
      bar_background: "#222",
      bar_foreground: "linear-gradient(to right, red, violet)",
      entities: ["timer.test2"],
    },
  ]);
  const card = dashboard.cards[0];

  await hass.callService("timer", "start", { duration: "00:00:10" }, { entity_id: "timer.test2" });
  await waitForTimerTime(card, "00:00:08");
  await expect(card).toMatchDualSnapshot("running");
});


it("Receding progress bar", async () => {
  const dashboard = await hass.Dashboard([
    {
      type: "custom:timer-bar-card",
      bar_direction: "rtl",
      bar_foreground: "#eee",
      bar_background: "var(--mdc-theme-primary, #6200ee)",
      entities: ["timer.test3"],
    },
  ]);
  const card = dashboard.cards[0];

  await hass.callService("timer", "start", { duration: "00:00:10" }, { entity_id: "timer.test3" });
  await waitForTimerTime(card, "00:00:08");
  await expect(card).toMatchDualSnapshot("running");
});
