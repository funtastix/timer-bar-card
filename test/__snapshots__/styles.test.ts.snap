// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors and icons: running 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card raised=\\"false\\">
    <slot></slot>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <div class=\\"generic-entity-row\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-state-icon data-domain=\\"timer\\" data-state=\\"active\\">…</ha-state-icon>
          </state-badge>
          <div class=\\"info pointer\\" title=\\"test2\\">
            test2
          </div>
          <div class=\\"bar-container pointer\\">
            <div class=\\"bar\\">
              <div></div>
            </div>
          </div>
          <div class=\\"text-content\\">
            8
          </div>
        </div>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;

exports[`Receding progress bar: running 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card raised=\\"false\\">
    <slot></slot>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <div class=\\"generic-entity-row\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-state-icon data-domain=\\"timer\\" data-state=\\"active\\">…</ha-state-icon>
          </state-badge>
          <div class=\\"info pointer\\" title=\\"test3\\">
            test3
          </div>
          <div class=\\"bar-container pointer\\">
            <div class=\\"bar\\">
              <div></div>
            </div>
          </div>
          <div class=\\"text-content\\">
            8
          </div>
        </div>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;

exports[`Widths: idle 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card raised=\\"false\\">
    <slot></slot>
    <h1 class=\\"card-header\\">
      Progress bar should be munched
    </h1>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <div class=\\"generic-entity-row\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-state-icon data-domain=\\"timer\\" data-state=\\"idle\\">…</ha-state-icon>
          </state-badge>
          <div class=\\"info pointer\\" title=\\"test1\\">
            test1
          </div>
          <div class=\\"text-content\\">
            Idle
          </div>
        </div>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;

exports[`Widths: running 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card raised=\\"false\\">
    <slot></slot>
    <h1 class=\\"card-header\\">
      Progress bar should be munched
    </h1>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <div class=\\"generic-entity-row\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-state-icon data-domain=\\"timer\\" data-state=\\"active\\">…</ha-state-icon>
          </state-badge>
          <div class=\\"info pointer\\" title=\\"test1\\">
            test1
          </div>
          <div class=\\"bar-container pointer\\">
            <div class=\\"bar\\">
              <div></div>
            </div>
          </div>
          <div class=\\"text-content\\">
            9:59:59
          </div>
        </div>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;
